packages/paste-core/utilities/box/src/FilterBoxPropsFromProps.ts

```diff
@@ -0,0 +1,101 @@
+import * as lodash from 'lodash';
+
+const supportedBoxProps = [
+  'margin',
+  'm',
+  'marginTop',
+  'mt',
+  'marginRight',
+  'mr',
+  'marginBottom',
+  'mb',
+  'marginLeft',
+  'ml',
+  'marginX',
+  'mx',
+  'marginY',
+  'my',
+  'padding',
+  'p',
+  'paddingTop',
+  'pt',
+  'paddingRight',
+  'pr',
+  'paddingBottom',
+  'pb',
+  'paddingLeft',
+  'pl',
+  'paddingX',
+  'px',
+  'paddingY',
+  'py',
+  'width',
+  'height',
+  'display',
+  'minWidth',
+  'minHeight',
+  'maxWidth',
+  'maxHeight',
+  'size',
+  'verticalAlign',
+  'overflow',
+  'overflowX',
+  'overflowY',
+  'alignItems',
+  'alignContent',
+  'justifyItems',
+  'justifyContent',
+  'flexWrap',
+  'flexDirection',
+  'flex',
+  'flexGrow',
+  'flexShrink',
+  'flexBasis',
+  'justifySelf',
+  'alignSelf',
+  'order',
+  'backgroundImage',
+  'backgroundSize',
+  'backgroundPosition',
+  'backgroundRepeat',
+  'backgroundColor',
+  'border',
+  'borderWidth',
+  'borderStyle',
+  'borderColor',
+  'borderRadius',
+  'borderTop',
+  'borderTopWidth',
+  'borderTopStyle',
+  'borderTopColor',
+  'borderTopLeftRadius',
+  'borderTopRightRadius',
+  'borderRight',
+  'borderRightWidth',
+  'borderRightStyle',
+  'borderRightColor',
+  'borderBottom',
+  'borderBottomWidth',
+  'borderBottomStyle',
+  'borderBottomColor',
+  'borderBottomLeftRadius',
+  'borderBottomRightRadius',
+  'borderLeft',
+  'borderLeftWidth',
+  'borderLeftStyle',
+  'borderLeftColor',
+  'borderX',
+  'borderY',
+  'borderColor',
+  'boxShadow',
+  'zIndex',
+] as const;
```

Comment:
As we add components we will want to have a single source of truth so we can keep things correct I would prefer we compose this list with things like...

```js
const boxProps = [
...spacingProps,
...borderProps,
...positionProps,
...alignmentProps
]
```

That way we can control all of the props components need from a singular place and we can derive types from it as well.

Reply:
Kind of makes me think of this: https://github.com/primer/components/blob/master/src/constants.js

Reply:
This was also copied from Primer back into Styled System https://styled-system.com/api/